src/vulkan-renderer/wrappers/instance.cpp

```diff
@@ -0,0 +1,174 @@
+        }
+    }
+
+    std::vector<const char *> instance_layers_wishlist = {};
+
+#ifndef NDEBUG
+    // RenderDoc is a very useful open source graphics debugger for Vulkan and other APIs.
+    // Not using it all the time during development is fine, but as soon as something crashes
+    // you should enable it, take a snapshot and look up what's wrong.
+    if (enable_renderdoc_instance_layer) {
+        instance_layers_wishlist.push_back("VK_LAYER_RENDERDOC_Capture");
+    }
+
+    // We can't stress enough how important it is to use validation layers during development!
+    // Validation layers in Vulkan are in-depth error checks for the application's use of the API.
+    // They check for a multitude of possible errors. They can be disabled easily for releases.
+    // Understand that in contrary to other APIs, in Vulkan API the driver provides no error checks
+    // for you! If you use Vulkan API incorrectly, your application will likely just crash.
+    // To avoid this, you must use validation layers during development!
+    if (enable_validation_layers) {
+        spdlog::debug("Vulkan validation layers are enabled.");
+        spdlog::debug("Adding 'VK_LAYER_KHRONOS_validation' to instance extension wishlist.");
+        instance_layers_wishlist.push_back("VK_LAYER_KHRONOS_validation");
+    } else {
+        spdlog::warn("Vulkan validation layers are not enabled although debug configuration is selected!");
+        spdlog::warn("You must always use validation layers during development if you are serious about writing stable software.");
+        spdlog::warn("Without them, most bugs are literally impossible to find.");
+        spdlog::warn("Even worse, your software might work on your machine but does crash on other systems!");
+    }
+#endif
+
+    // Add requested instance layers to wishlist.
+    if (requested_instance_layers.has_value()) {
+        for (const auto &instance_layer : requested_instance_layers.value()) {
+            instance_layers_wishlist.push_back(instance_layer.c_str());
+        }
+    }
+
+    // We have to check which instance layers of our wishlist are available on the current system!
+    // We are not checking for duplicated entries but this is no problem.
+    for (auto current_layer : instance_layers_wishlist) {
+        if (availability_checks->has_instance_layer(current_layer)) {
+            spdlog::debug("Adding '{}' to list of enabled instance layers.", current_layer);
+            enabled_instance_layers.push_back(current_layer);
+        } else {
+#ifdef NDEBUG
+            if (std::string(current_layer).compare(VK_EXT_DEBUG_MARKER_EXTENSION_NAME) == 0) {
+                spdlog::error("You can't use command line argument -renderdoc in release mode.");
+            }
+#else
+            spdlog::error("Requested instance layer '{}' is not available on this system!", current_layer);
+#endif
+        }
+    }
+
+    VkInstanceCreateInfo instance_create_info = {};
```

Comment:
Maybe this could just be called `create_info`? Since we're already in the `Instance` class.

Reply:
Yes.

Reply:
done.